src/Discord/functions.php

```diff
@@ -38,3 +38,15 @@ function mentioned($part, Message $message)
         return strpos($message->content, $part) !== false;
     }
 }
+
+/**
+ * Calculates UNIX timestamp from snowflake.
+ *
+ * @param STRING $snowflake The snowflake
+ *
+ * @return UNIX timestamp
+ */
+function timestampFromSnowFlake(string $snowflake)
+{
+    return (($snowflake >> 22) + 1420070400000);
```

Comment:
It returns a timestamp with microseconds, so to resolve the UNIXtime, you get the Timestamp w/o MicroSeconds you'd do floor($timestamp / 1000). I have yet to find a way to resolve the Timestamp w/ MicroSeconds using Carbon though.. (I used http://i.imgur.com/UxWvdYD.png as a guide)